js.go

```diff
@@ -826,6 +863,67 @@ type jsSub struct {
 	cmeta atomic.Value
 }
 
+// newFetchReply generates a unique inbox used for a fetch request.
+func (jsi *jsSub) newFetchReply() string {
+	jsi.mu.Lock()
+	rpre := jsi.rpre
+	rn := jsi.rr.Int63()
+	jsi.mu.Unlock()
+	var sb strings.Builder
+	sb.WriteString(rpre)
+	var b [aReplyTokensize]byte
+	for i, l := 0, rn; i < len(b); i++ {
+		b[i] = rdigits[l%base]
+		l /= base
+	}
+	sb.Write(b[:])
+	return sb.String()
+}
+
+// handleFetch is delivered a message requested by pull subscribers
+// when calling Fetch.
+func (jsi *jsSub) handleFetch(m *Msg) {
+	jsi.mu.Lock()
+	if len(jsi.freqs) > 0 {
+		mch := jsi.freqs[0]
```

Comment:
This pattern will bite us ;)

Reply:
I wonder if this would work better as a channel of channels...

Reply:
Let's do the append trick we discussed.

Reply:
And add to ping/pong as needed.

Reply:
Sending different PR for the ping/pong change

Reply:
https://github.com/nats-io/nats.go/pull/723